api/src/main/java/org/openmrs/PersonName.java

```diff
@@ -145,22 +146,28 @@ public static PersonName newInstance(PersonName pn) {
 		}
 		PersonName newName = new PersonName(Integer.valueOf(pn.getPersonNameId()));
 		if (pn.getGivenName() != null) {
-			newName.setGivenName(String.valueOf(pn.getGivenName()));
+			String givenNameAftreTrim = pn.getGivenName().trim();
+			newName.setGivenName(String.valueOf(givenNameAftreTrim));
```

Comment:
Why do you have to call String.valueOf?

Reply:
'String.valueOf' part was there in the original code, and i didn't change it. I don't know the exact purpose of that